src/main/java/organice/model/ModelManager.java

```diff
@@ -284,9 +284,24 @@ public void matchDonors(Patient patient) {
     }
 
     /**
-     * Retrieves the Match List
+     * Returns a copy of the match list.
      */
     public ObservableList<Person> getMatchList() {
-        return listOfMatches;
+        ObservableList<Person> listOfMatchesCopy = FXCollections.observableArrayList();
```

Comment:
Are `patient` and `donor` the only two possible `persons` in `listOfMatches`? If no, might wanna add a guard clause to prevent `doctors` etc from being accidentally typecast and added in the else block

Reply:
Added